### ISSUE TITLE
Make the default setting for curves Y = X instead of Y = 0.

### ISSUE THREAD

**Issue by Miami-Mike** [EdgeTX/edgetx#1067]
This went stale after [6 months on the OpenTX forum](https://github.com/opentx/opentx/issues/8492#issuecomment-970224829) but I think it's a good idea:

Instead of the default for curves being Y = 0% regardless of X, make it Y = X. That is, instead of a horizontal line on the X axis, make it a diagonal line from (X = -100%, Y = -100%) to (X = +100%, Y = +100%). That way, if a curve is applied to an input, mix, or output without first defining it, it won't cause unintended model behavior such as a motor suddenly turning on. This is already what you get by clicking "Apply" with the Curve Creator at its default settings, so why not make it the curve default?

- [ ] Radio side
- [ ] Companion side (Resolved by #1078)

**Comment by pfeerick** [comment]
Sounds like an excellent idea... the default "curve" really should be... well, normal stick movement... 🤦

**Comment by raphaelcoeffic** [comment]
@Miami-Mike i had the feeling I had done exactly this already. But maybe this was in a dream.

**Comment by Miami-Mike** [comment]
[You commented about it in the OpenTX thread](https://github.com/opentx/opentx/issues/8492#issuecomment-835947616) on May 9th.

**Comment by pfeerick** [comment]
Wait a minute... "Curve creator" ... you're talking about Companion, not the radio, aren't you? As the radio firmware defaults to creating a -100 => 100 curve by default.

**Comment by Miami-Mike** [comment]
I probably shouldn't have mentioned the Curve Creator because it really doesn't have anything to do with this.

If you make it so that a curve can't be selected until it's defined then that might solve the issue that chronowarper and Raphael mentioned, but it could be a bit inconvenient. If you're setting up an input, mix, or output and you want to include a curve, you might find it annoying that the system won't let you do that. You'd first have to interrupt what you're doing, go to the Curves page, define the curve, and then go back to where you were.

I think the way I suggested would work out better because a Y = X curve could be included in any input, mix, or output, but would have no effect. It would be essentially transparent. You'd be able to first finish setting up the line you're working on and then proceed to the Curves page to adjust the curve.

**Comment by elecpower** [comment]
As an aside, the ability to edit a curve from a mix or input is on my long todo list

**Comment by elecpower** [comment]
I can knock up the Companion PR for Y = X if we are in agreement. Thumbs up/down @raphaelcoeffic @pfeerick

**Comment by Miami-Mike** [comment]
Don't forget that when you do _Print Model_ in Companion, curves with all Y=0 values are presumably considered unused and don't show up in the printout. You'll want to change that so that instead, curves with all Y=X values are the ones omitted from printouts.

**Comment by elecpower** [comment]
Tested and Y=X now not shown

**Comment by Eldenroot** [comment]
> Sounds like an excellent idea... the default "curve" really should be... well, normal stick movement... 🤦 ... if I get a chance in the next day or so, I'll look at it...

This means that the curve editing would be more newbie friendly?

**Comment by pfeerick** [comment]
No, it has nothing to do with that. It would mean, as is already stated in the first post, that the *default* curve would be no different normal stick movement, rather than a flat 50% as it is now on radio.

**Comment by raphaelcoeffic** [comment]
Sort of: when creating a new curve on radio, it will default to `f(x)=x`, however, if you just select some unused curve in mixer, input or outputs, it will use `f(x)=0`.

**Comment by gagarinlg** [comment]
Since the curve page now only displays configured curves, maybe the curve selection pup-up could do the same?

**Comment by pfeerick** [comment]
You'll get complainers either way - those who select the curve as they setup the mixes, and then go setup up the curves afterwards, and then those who say it should only show configured curves. :shrug: It's no different with GVs... ie. use GV1, GV2 and then go setup the parameters for the GV afterwards. Or do you setup up GV1 and GV2, and then configure the mixes, LS etc?
